### PR TITLE
Add Start time entry view model

### PR DESCRIPTION
**Pull request by willsb**
Very basic stuff, just to allow navigation. The whole logic for parsing raw text and actual creation of the time entry will come later.

Squash whenever

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+namespace Toggl.Foundation.MvvmCross.Parameters
+{
+    public sealed class DateParameter
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs]: Why not internal?

File: Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+        public static DateParameter WithDate(DateTimeOffset date) => new DateParameter
+        {
+            DateString = date.ToString()
+        };
+    }
+
+    internal static class DateParameterExtensions
+    {
+        public static DateTimeOffset GetDate(this DateParameter self)
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs]: Why is this an extension method an not just a regular method on the `DateParameter` class?

File: Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs]: Missing newline between `using` and `namespace`.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/StartTimeEntryViewModelTests.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+{
+    public class StartTimeEntryViewModelTests
+    {
+        public class StartTimeEntryViewModelTest : BaseViewModelTests<StartTimeEntryViewModel>
+        {
+            protected override StartTimeEntryViewModel CreateViewModel()
+                => new StartTimeEntryViewModel(DataSource, NavigationService);
+        }
+
+        public class TheConstructor : StartTimeEntryViewModelTest
```
Comment (arturs105) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/StartTimeEntryViewModelTests.cs]: This doesn't really need to inherit from `StartTimeEntryViewModelTest`.

File: Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs (hunk 1 above)
Comment (willsb) [inline Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs]: MvvmCross needs to be able to create those

File: Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs (hunk 2 above)
Comment (willsb) [inline Toggl.Foundation.MvvmCross/Parameters/DateParameter.cs]: The class can't have any methods. MvvmCross, again

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/StartTimeEntryViewModelTests.cs (hunk 4 above)
Comment (willsb) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/StartTimeEntryViewModelTests.cs]: It does. It uses the `NavigationService` and the `DataSource`

**Pull request merged**